packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+ * @param n - The bigint to convert.
+ * @returns The buffer representation of the bigint.
+ */
+export function bigintToBuffer(n: bigint): Buffer {
+    return beBigintToBuffer(n)
+}
+
+/**
+ * Converts a BigNumberish type to a buffer, handling various input types and converting them to bigint first if necessary.
+ * @param n The BigNumberish value to convert.
+ * @returns The buffer representation of the BigNumberish value.
+ */
+export function beBigNumberishToBuffer(n: BigNumberish): Buffer {
+    if (
+        typeof n === "number" ||
+        typeof n === "bigint" ||
+        (typeof n === "string" && isStringifiedBigint(n)) ||
+        (typeof n === "string" && isHexadecimal(n))
+    ) {
+        return bigintToBuffer(BigInt(n))
+    }
+
+    return n as Buffer
+}
+
+/**
+ * Converts a BigNumberish type to a buffer. Alias for beBigNumberishToBuffer.
+ * @param n The BigNumberish value to convert.
+ * @returns The buffer representation of the BigNumberish value.
+ */
+export function bigNumberishToBuffer(n: BigNumberish): Buffer {
+    return beBigNumberishToBuffer(n)
+}
+
+/**
+ * Converts a BigNumberish type to a bigint, handling various input types.
+ * @param n The BigNumberish value to convert.
+ * @returns The bigint representation of the BigNumberish value.
+ */
+export function beBigNumberishToBigint(n: BigNumberish): bigint {
+    if (
+        typeof n === "number" ||
+        typeof n === "bigint" ||
+        (typeof n === "string" && isStringifiedBigint(n)) ||
+        (typeof n === "string" && isHexadecimal(n))
+    ) {
+        return BigInt(n)
+    }
+
+    return bufferToBigint(n as Buffer)
+}
+
+/**
+ * Converts a BigNumberish type to a bigint. Alias for beBigNumberishToBigint.
+ * @param n The BigNumberish value to convert.
+ * @returns The bigint representation of the BigNumberish value.
+ */
+export function bigNumberishToBigint(n: BigNumberish): bigint {
+    return beBigNumberishToBigint(n)
+}
+
+/**
+ * Converts a buffer to a bigint assuming little-endian byte order.
+ * @param buffer The buffer to convert.
+ * @returns The bigint representation of the buffer's contents in little-endian.
+ */
+export function leBufferToBigint(buffer: Buffer): bigint {
+    return BigInt(`0x${Buffer.from(buffer).reverse().toString("hex")}`)
+}
+
+/**
+ * Converts a bigint to a buffer, assuming little-endian byte order, and fills with zeros if necessary.
+ * @param n The bigint to convert.
+ * @param length The number of bytes of the buffer to return.
+ * @returns The buffer representation of the bigint in little-endian.
+ */
+export function leBigintToBuffer(n: bigint, length?: number): Buffer {
+    const hex = bigIntToHexadecimal(n)
+
+    // Calculate the minimum buffer length required to represent 'n' in bytes.
+    // Each hexadecimal character represents 4 bits, so 2 characters are 1 byte.
+    const minLength = Math.ceil(hex.length / 2)
+
+    // Use the provided length or the calculated minimum length, whichever is greater.
```

Comment:
Similar to above, this behavior should be documented, and I have a similar question as to whether it's better to rename length -> minLength, or to throw on overflow.